drivers/sensor/lis2mdl/lis2mdl.c

```diff
@@ -261,6 +331,9 @@ static int lis2mdl_init_interface(const struct device *dev)
 
 static const struct lis2mdl_config lis2mdl_dev_config = {
 	.master_dev_name = DT_INST_BUS_LABEL(0),
+	.single_mode = DT_INST_NODE_HAS_PROP(0, single_mode),
+	.cancel_offset = DT_INST_NODE_HAS_PROP(0, cancel_offset),
```

Comment:
More details (I wanted to put the -1 in as quickly as I could to avoid this getting merged):

DT_INST_NODE_HAS_PROP should never be used for boolean properties. DT_PROP (and the DT_INST_PROP convenience helper) do the right thing with booleans:

https://docs.zephyrproject.org/latest/reference/devicetree/api.html#c.DT_PROP

```
For properties whose bindings have the following types, this macro expands to:
[...]
    boolean: 0 if the property is false, or 1 if it is true
```